src/tests/ftest/network/zero_config.py

```diff
@@ -0,0 +1,161 @@
+
+    Test Class Description:
+        Test to verify that client application to libdaos can access a running
+        DAOS system with & without any special environment variable definitions.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a ZeroConfigTest object."""
+        super(ZeroConfigTest, self).__init__(*args, **kwargs)
+        self.setup_start_servers = False
+
+    def get_port_cnt(self, hosts, dev, port_counter):
+        """Get the port count info for device names specified.
+
+        Args:
+            hosts (list): list of hosts
+            dev (str): device to get counter information for.
+            port_counter (str): port counter to get information from
+
+        Returns:
+            dict: a dictionary of data values for each NodeSet key
+
+        """
+        b_path = "/sys/class/infiniband/{}".format(dev)
+        file = os.path.join(b_path, "ports/1/counters", port_counter)
+
+        # Check if if exists for the host
+        check_result = check_file_exists(hosts, file)
+        if not check_result[0]:
+            self.fail("{}: {} not found".format(check_result[1], file))
+
+        cmd = "cat {}".format(file)
+        text = "port_counter"
+        error = "Error obtaining {} info".format(port_counter)
+        return get_host_data(hosts, cmd, text, error, 20)
+
+    @fail_on(CommandFailure)
+    def verify_client_run(self, server_idx, exp_iface, env):
+        """Verify the interface assigned by running a libdaos client.
+
+        Args:
+            server_idx (int): server to get config from.
+            exp_iface (str): expected interface to check.
+            env (bool): add OFI_INTERFACE variable to exported variables of
+                client command.
+
+        Returns:
+            bool: returns status
+
+        """
+        hfi_map = {"ib0": "hfi1_0", "ib1": "hfi1_1"}
+
+        # Get counter values for hfi devices before and after
+        cnt_before = self.get_port_cnt(
+            self.hostlist_clients, hfi_map[exp_iface], "port_rcv_data")
+
+        # Let's run daos_racer as a client
+        daos_racer = DaosRacerCommand(self.bin, self.hostlist_clients[0])
+        daos_racer.get_params(self)
+
+        # Update env_name list to add OFI_INTERFACE if needed.
+        if env:
+            daos_racer.update_env_names(["OFI_INTERFACE"])
+        daos_racer.set_environment(
+            daos_racer.get_environment(self.server_managers[server_idx]))
+        daos_racer.run()
+
+        # Verify output and port count to check what iface CaRT init with.
+        cnt_after = self.get_port_cnt(
+            self.hostlist_clients, hfi_map[exp_iface], "port_rcv_data")
+
+        diff = 0
+        for cnt_b, cnt_a in zip(cnt_before.values(), cnt_after.values()):
+            diff = int(cnt_a) - int(cnt_b)
+            self.log.info("Port [%s] count difference: %s", exp_iface, diff)
+
+        # If we don't see data going through the device, fail
```

Comment:
Are there any utilities that run in CI that might send network traffic over any of these interfaces?  If so, then it would generate a false positive 'success' because the data counters would be > 0.  If there's nothing else, even a ping, then this should be safe.  I just want to make sure that the 'use detection' is valid.  You can grep the daos.log to see if the client overrides were used.  For example, if you ```export OFI_INTERFACE=myfakedevice``` you will see a corresponding entry in daos.log that says ```07/10-19:03:47.91 wolf-77 DAOS[54738/54738] mgmt INFO src/mgmt/cli_mgmt.c:503 dc_mgmt_net_cfg() Using client provided OFI_INTERFACE: myfakedevice```  Using the counters is good, if it's reliable though.

Reply:
This is a good point. I know you had brought this up before and I had forgotten about it, my apologies. I can definitely add something like that to the test and like you mentioned, avoid false positives.